src/librustdoc/html/static/js/search.js

```diff
@@ -1671,6 +1591,50 @@ function initSearch(rawSearchIndex) {
             }
             const maxEditDistance = Math.floor(queryLen / 3);
 
+            /**
+             * Convert names to ids in parsed query elements.
+             * This is not used for the "In Names" tab, but is used for the
+             * "In Params", "In Returns", and "In Function Signature" tabs.
+             *
+             * If there is no matching item, but a close-enough match, this
+             * function also that correction.
+             *
+             * See `buildTypeMapIndex` for more information.
+             *
+             * @param {QueryElement} elem
+             */
+            function convertNameToId(elem) {
```

Comment:
So I understand correctly, if multiple items give the same distance, we pick the first one we encountered right? I'm not sure if it's a good idea either. Not against it, just not sure.

Reply:
That's already a nice improvement. :+1: 

The problem here in my opinion is that it'll still "not make sense" for users. I think just rewording the sentence could help with it? So currently we have:

```
Type "streaming" not found. Showing results for "string" instead.
```

What do you think instead of:

```
Type "streaming" not found. Showing results for closest type name "string" instead.
```

?